packages/stark-ui/src/modules/progress-indicator/directives/progress-indicator.directive.ts

```diff
@@ -0,0 +1,92 @@
+import {
+	ComponentFactory,
+	ComponentFactoryResolver,
+	ComponentRef,
+	Directive,
+	ElementRef,
+	Inject,
+	Input,
+	OnDestroy,
+	OnInit,
+	Renderer2,
+	ViewContainerRef
+} from "@angular/core";
+import { STARK_PROGRESS_INDICATOR_SERVICE, StarkProgressIndicatorService } from "../services";
+import { StarkProgressIndicatorConfig, StarkProgressIndicatorType } from "../entities";
+import { Subscription } from "rxjs";
+import { StarkProgressIndicatorComponent } from "../components/progress-indicator.component";
+
+/**
+ * Name of the directive
+ */
+const directiveName: string = "[starkProgressIndicator]";
+
+@Directive({
+	selector: directiveName
+})
+export class StarkProgressIndicatorDirective implements OnInit, OnDestroy {
```

Comment:
Can you add a JSDoc for this and explain how this directive works (registers itself, dynamically creates a component, etc... )